engine/robot.py

```diff
@@ -360,98 +375,131 @@ def track_obstacle(self):
                 try:
                     line = content[counter]
                     counter += 1
-                    curr_ultrasonic_value = float((''.join(line.rstrip('\n')).strip('()').split(', '))[0])
+                    curr_ultrasonic_value = float(
+                        (''.join(line.rstrip('\n')).strip('()').split(', '))[0])
                 except IndexError:
                     print("no more sensor data")
                     break
             else:
-                curr_ultrasonic_value = self.robot_state.front_ultrasonic.distance()
+                from electrical.ultrasonic_sensor import Ultrasonic
```

Comment:
what do you mean by adding the is_sim flag check there?

Reply:
also, I can't move the import to be global bc that make the whole robot class "no longer simulating". TLDR, import causes code to not pass test bc import relies on IRL port initialization, which is not true

Reply:
So is it not possible to do a is_sim flag check at the very beginning of the file, and determine if we can import Ultrasonic there? I remember we did something very similar for robot_state.py

Reply:
oh, I see

Reply:
hm, I don't think I can. robot_state, which has the is_sim flag, is initialized in __init__, so I can't check if not is_sim at the top of the file because it is out of scope

Reply:
oh!